# 4.6 Combining and Exporting Exercise

In [3]:
#Import Pandas Numpy and OS
import pandas as pd
import numpy as np
import os

# 3. In a new notebook, import the orders_products_combined dataframe from the pickle file you just saved.

In [6]:
df_merged_large = pd.read_pickle('/Users/ariedlin/Desktop/Career Foundry Assignments/8-24 Instacart Blanket Analysis Real/2 Data/2 Prepared Data/df_merged_large')

# 4. Check the shape of the imported dataframe (it should be the same as the one you exported—always check!).

In [9]:
df_merged_large.shape

(32434489, 12)

This matches the 32,434,489 and 12 from the previous notebook.

# 5. Determine a suitable way to combine the orders_products_combined dataframe with your products data set.

In [11]:
#Import products dataframe
df_prods_clean_no_dups = pd.read_csv('/Users/ariedlin/Desktop/Career Foundry Assignments/8-24 Instacart Blanket Analysis Real/2 Data/2 Prepared Data/df_prods_clean_no_dups.csv')

In [14]:
#Check shape of df_prods_clean_no_dups
df_prods_clean_no_dups.shape

(49672, 6)

In [18]:
#Use head to review columns in df_prods_clean_no_dups dataframe
df_prods_clean_no_dups.head()

Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [20]:
#Use head function to review columns in df_merged_large dataframe
df_merged_large.head()

Unnamed: 0  order_id  user_id eval_set  order_number  orders_dow  \
0           0   2539329        1    prior             1           2   
1           0   2539329        1    prior             1           2   
2           0   2539329        1    prior             1           2   
3           0   2539329        1    prior             1           2   
4           0   2539329        1    prior             1           2   

   order_hour_of_day  days_since_prior_order  product_id  add_to_cart_order  \
0                  8                     NaN         196                  1   
1                  8                     NaN       14084                  2   
2                  8                     NaN       12427                  3   
3                  8                     NaN       26088                  4   
4                  8                     NaN       26405                  5   

   reordered _merge  
0          0   both  
1          0   both  
2          0   both  
3          0   both  
4          0   both

In [28]:
#Use df_merge function to combine df_prods_clean_no_dups and df_merged_large using product_id column to link them
df_ords_prods_prior_merged = df_merged_large.merge(df_prods_clean_no_dups, on='product_id', indicator='merge_status')

# 6. Confirm the results of the merge using the merge flag.

In [30]:
#Check to make sure the merge looks correct
df_ords_prods_prior_merged.head()

Unnamed: 0_x  order_id  user_id eval_set  order_number  orders_dow  \
0             0   2539329        1    prior             1           2   
1             0   2539329        1    prior             1           2   
2             0   2539329        1    prior             1           2   
3             0   2539329        1    prior             1           2   
4             0   2539329        1    prior             1           2   

   order_hour_of_day  days_since_prior_order  product_id  add_to_cart_order  \
0                  8                     NaN         196                  1   
1                  8                     NaN       14084                  2   
2                  8                     NaN       12427                  3   
3                  8                     NaN       26088                  4   
4                  8                     NaN       26405                  5   

   reordered _merge  Unnamed: 0_y                             product_name  \
0          0   both           195                                     Soda   
1          0   both         14084  Organic Unsweetened Vanilla Almond Milk   
2          0   both         12427                      Original Beef Jerky   
3          0   both         26089               Aged White Cheddar Popcorn   
4          0   both         26406         XL Pick-A-Size Paper Towel Rolls   

   aisle_id  department_id  prices merge_status  
0        77              7     9.0         both  
1        91             16    12.5         both  
2        23             19     4.4         both  
3        23             19     4.7         both  
4        54             17     1.0         both

In [32]:
#Check Shape
df_ords_prods_prior_merged.shape

(32404859, 18)

In [34]:
#Check value counts
df_ords_prods_prior_merged['_merge'].value_counts()

_merge
both          32404859
left_only            0
right_only           0
Name: count, dtype: int64

# 7. Export the newly created dataframe as ords_prods_merge in a suitable format (taking into consideration the size).

In [39]:
#Export df_ords_prods_prior_merged as a pickle
df_ords_prods_prior_merged.to_pickle('/Users/ariedlin/Desktop/Career Foundry Assignments/8-24 Instacart Blanket Analysis Real/2 Data/2 Prepared Data/df_ords_prods_prior_merged')